cmake/SundialsSetupCompilers.cmake

```diff
@@ -223,13 +223,6 @@ if(SUNDIALS_POSIX_TIMERS AND POSIX_TIMERS_NEED_POSIX_C_SOURCE)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_POSIX_C_SOURCE=${SUNDIALS_POSIX_C_SOURCE}")
 endif()
 
-# Check if profiling is being built with no timers.
-if(SUNDIALS_BUILD_WITH_PROFILING AND
-   (NOT ENABLE_CALIPER) AND
-   (NOT ENABLE_MPI) AND
-   (NOT SUNDIALS_POSIX_TIMERS))
-  message(SEND_ERROR "The SUNDIALS native profiler requires POSIX timers or MPI_Wtime, but neither were found.")
-endif()
```

Comment:
Do we always build the internal profiler, even when Caliper is enabled? If so, we should double check that the documentation notes POSIX or windows timers are required when profiling is enabled and remove potential the MPI requirement if that was noted.

Reply:
Yes, we include the profiler even when caliper is enabled because otherwise user code would need a lot of ifdefs. I'll add a note to the docs.